react-validated-input-component/src/validated-input.jsx

```diff
@@ -17,7 +16,11 @@ class ValidatedInput extends React.Component {
   handleChange(event) {
     const value = event.target.value;
     const name = event.target.name;
-    this.setState({ [name]: value });
+    const regex = /^(?=.*\d)(?=.*[A-Z])(?=.*[a-zA-Z]).+$/gm;
+    const result = value.match(regex);
+    if (result) {
+      this.setState({ [name]: value });
+    }
```

Comment:
This is looking good, but you want to set an error message in state if the password isn't valid. You can then use the error message to display to the user so they know how to fix it.